### ISSUE TITLE
tbkeys-lite commands for shortcuts

### ISSUE THREAD

**Issue by zwergneun** [thunderbird/import-export-tools-ng#519]
Hi there,

i would like to add an shortcut to tb for exporting a folder to mbox but i can't find the commands.
Would be nice if somebody could help.
The way with mouse and menu is long and time-consuming.
Thanks

**Comment by cleidigh** [comment]
@zwergneun 
Thanks for the post.
Currently the folder operations cannot be called without dynamic parameters from the new menu system. Yes I know the menus can e deep, hard to otherwise with so many options.

That said, I think I can do similar tweaks as I did for ExportSelectedMessages, also to support shortcuts.
If you are still interested I will take a look.
@cleidigh

**Comment by zwergneun** [comment]
@cleidigh 
Thanks for the answer.
Im still interested, but only if it isn't to complicated for u.
I think shortcuts are the best way to safe time (working as an architect i could not use my CAD software efficient without them).
I have only small experience in programming so i thought if you add an add-on there has to be commands for thunderbird similar to the commands here:
https://hg.mozilla.org/comm-central/file/tip/mail/base/content/mainCommandSet.inc.xhtml
And it would be easy to call them with tbkeys-lite, if i knew the commands.

**Comment by cleidigh** [comment]
@zwergneun 
I totally agree about shortcuts. I always used them for my schematic and PCB CAD work!

I have a beta for you to try. It allows shortcuts for mbox export command.
https://github.com/thunderbird/import-export-tools-ng/blob/v14.0.2/xpi/beta/import-export-tools-ng-14.0.2-b6-tb.xpi

The function is
exportfolder()
It takes an object parameter with the following properties : 

selectedFolder - always '//currentFolder'
localFolder - always true
includeSubfolders - true/false
flattenSubfolders - true/false
zip- true/false

Here is the example for an individual mbox export:

```

exportfolder({selectedFolder: '//curentFolder', localFolder: true, zip: false, includeSubfolders: false, flattenSubfolders: false})

```

I have tested with ImportExportTools built in shortcut capabilities. 
Let me know how you fare...
@cleidigh

**Comment by cleidigh** [comment]
@zwergneun 
Did you get my post?
@cleidigh

**Comment by zwergneun** [comment]
@cleidigh
Sorry for not responding, too much work to do.
I downloaded the file and installed it.
How do i have to use the function? 
I really don't understand the ImportExportTools built in shortcut capabilities.
Thanks so far.

**Comment by cleidigh** [comment]
@zwergneun 
Apologies, I was not clear, you don't need to use the internal shortcuts. I asume you are already using TBkeys so best to just use that. Define your key and use the function signature from my Examples above. Just note exact syntax.
@cleidigh

**Comment by zwergneun** [comment]
@cleidigh
I tested it with tbkeys-lite, this is the code.
`"e": "exportfolder({selectedFolder: '//curentFolder', localFolder: true, zip: false, includeSubfolders: false, flattenSubfolders: false})"`
When i select a folder and press e nothing is happening.

**Comment by cleidigh** [comment]
@zwergneun 
hmm, you have the function call correct, including my stupid typo for currentFolder, however, the typo is in the code also.
I can try TBkeys tomorrow.
Try shift-control-x for the ke, maybe e is protected Just to match what I used.
@cleidigh

**Comment by zwergneun** [comment]
@cleidigh
I tried shift-control-x and also x nothing happening.
Thanks for your work.

**Comment by cleidigh** [comment]
@zwergneun 
Ok, I figured out the issues after directly testing tbKeys-lite.

So the first thing is that you can't use tbKeys-lite, you have to use the same author's tbKeys which supports the required function syntax. The author does not have this on the addon site, but the download is available on his support site here (grab tbkeys.xpi)
https://github.com/wshanks/tbkeys/releases

Save your keybindings so you can reconfigure after uninstall and installation.

Then you need to add window. To the function call. Here is my entry
```
"e": "window.exportfolder({selectedFolder: '//curentFolder', localFolder: true, zip: false, includeSubfolders: false, flattenSubfolders: false});"
```

@cleidigh

**Comment by zwergneun** [comment]
@cleidigh
Wow, works like a charm.
Thanks so much!
Thats a real time saver.
Is there any way you could give me the function for a pdf export? 
I like to archive some mails as pdf because i can directly search the mails without importing the mbox file before.

**Comment by cleidigh** [comment]
@zwergneun 
Great, glad it's working.
Are you talking about exporting selected messages or a folder?
@cleidigh

**Comment by zwergneun** [comment]
@cleidigh
Selected messages would be fine, if possible.
All messages in a folder is also fine.

**Comment by cleidigh** [comment]
@zwergneun 
I had to tweak the pdf functions to deal with shortcuts.

For pdf export use the following tbkeys entries : (first parameter is wether to export full folder or selected messages)
folder:
    "d": "window.IETprintPDFmain.print(true,{});"
selected messages : 
    "d": "window.IETprintPDFmain.print(false,{});",

Note: b7 corrects two errors in the mbox export call, you will have to edit your tbkeys entry.
//curentFolder needs to be changed to //currentFolder
The zip option changes to zipped

https://github.com/thunderbird/import-export-tools-ng/blob/v14.0.2/xpi/beta/import-export-tools-ng-14.0.2-b7-tb.xpi

@cleidigh

**Comment by cleidigh** [comment]
@zwergneun 
Great!
Reviews are always welcome.
@cleidigh
